node/rpc/src/api.rs

```diff
@@ -56,24 +58,27 @@ build_rpc_trait! {
     }
 }
 
-pub struct RpcImpl {
+pub struct RpcImpl<S: Signer> {
     state_db_viewer: StateDbViewer,
     submit_txn_sender: Sender<SignedTransaction>,
+    signer: Arc<S>,
 }
 
-impl RpcImpl {
+impl<S: Signer> RpcImpl<S> {
     pub fn new(
         state_db_viewer: StateDbViewer,
         submit_txn_sender: Sender<SignedTransaction>,
+        signer: Arc<S>
     ) -> Self {
-        RpcImpl {
+        Self {
             state_db_viewer,
             submit_txn_sender,
+            signer,
```

Comment:
this signing bit is incorrect, unless you have discussed otherwise with @ilblackdragon. the API is intended to work as already implemented.
1) you submit a request with a user friendly RPC interface (bs58 encodings, etc)
2) you get a response with a "PreparedTransactionBody" to sign. this uses non-user friendly decodings antd formats that may not be suitable for an external API, but you are still able to verify that it is a mutation of your original request.
3) the consumer signs it and then responds with the signature and body.